app/models/post.rb

```diff
@@ -57,6 +57,34 @@ def address
     [street, city, state, postal_code, country].compact.join(", ")
   end
 
+  def self.filter_type(type)
+    if type == "needed" || type == "available"
+      send(type.to_sym)
+    else
+      all
```

Comment:
It's just like calling `Post.all`, but since we're within `Post`, we can just say `all`. This lets us chain them. 
For example, you can say `Post.where(id: [1,2,3,4,5]).all.all.all.all` and it just returns the `where`.
http://apidock.com/rails/ActiveRecord/Scoping/Named/ClassMethods/all